src/lib/socks/headers.ts

```diff
@@ -484,3 +560,20 @@ export function interpretResponseBuffer(buffer:ArrayBuffer) : Response {
 
   return response;
 }
+
+// Helper methods for encoding strings to/from ArrayBuffer
+
+// ArrayBuffer to string
+function ab2str(buf:ArrayBuffer) : string {
```

Comment:
BTW, we can use this API:
https://nodejs.org/dist/latest-v4.x/docs/api/buffer.html#buffer_buf_tostring_encoding_start_end